File: part_2/regression_template.py
import warnings
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import PolynomialFeatures

warnings.filterwarnings(action="ignore", module="scipy", message="^internal gelsd")

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Fitting Regression to dataset
poly_reg = PolynomialFeatures(degree=4)
X_poly = poly_reg.fit_transform(X)
lin_reg = LinearRegression()
lin_reg.fit(X_poly, y)

# predicting a new result with Polynomial Regression
print(lin_reg.predict(poly_reg.fit_transform(6.5)))

# Visualization
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color="red")
plt.plot(X_grid, lin_reg.predict(poly_reg.fit_transform(X_grid)), color="blue")
plt.title("Truth or Bluff (Polynomial Regression)")
plt.xlabel("position level")
plt.ylabel("salary")
plt.show()

File: part_2/regression_template.R
# :help Nvim-R
# \rf - start r session
# \rq - close r session
# \l  — send current line
# \ae - Send file
# \se — send current selection (and echo)

library(caTools)
library(ggplot2)

# dataset = read.csv("Position_Salaries.csv")
# dataset = dataset[2:3]

# Fitting Polynomial Regression Model to the dataset
# For example:
# dataset$Level2 = dataset$Level^2
# dataset$Level3 = dataset$Level^3
# dataset$Level4 = dataset$Level^4
# regressor = lm(formula = Salary ~ .,
#              data = dataset)

# PREDICTION
# For example:
# yPred2 = predict(regressor, data.frame(Level = 6.5,
#                                        Level2 = 6.5^2,
#                                        Level3 = 6.5^3,
#                                        Level4 = 6.5^4))

# VISUALIZATION
# Polynomial Regression
# For Example:
# ggplot() +
#     geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = "red") +
#     geom_line(aes(x = dataset$Level, y = predict(regressor, newData = dataset)), colour = "blue") +
#     ggtitle("Truth or Bluff (Linear Regression)") +
#     xlab("position level") +
#     ylab("salary")

# polynomial with higher resolution
# xGrid = seq(min(dataset$Level), max(dataset$Level), 0.1)
# ggplot() +
#     geom_point(aes(x = dataset$Level ,y = dataset$Salary), colour="red") +
#     geom_line(aes(x = xGrid, y = predict(regressor,
#                                          newdata = data.frame(Level = xGrid,
#                                                               Level2 = xGrid^2,
#                                                               Level3 = xGrid^3,
#                                                               Level4 = xGrid^4))),
#               colour="blue") +
#     ggtitle("Truth or Bluff (Linear Regression)") +
#     xlab("position level") +
#     ylab("salary")
